x-pack/filebeat/input/awscloudwatch/cloudwatch.go

```diff
@@ -151,14 +154,30 @@ func (p *cloudwatchPoller) startWorkers(
 // equal time.Now) to allow deterministic unit tests.
 func (p *cloudwatchPoller) receive(ctx context.Context, logGroupIDs []string, clock func() time.Time) {
 	defer p.workerWg.Wait()
+
 	// startTime and endTime are the bounds of the current scanning interval.
-	// If we're starting at the end of the logs, advance the start time to the
-	// most recent scan window
-	var startTime time.Time
 	endTime := clock().Add(-p.config.Latency)
-	if p.config.StartPosition == "end" {
+
+	var startTime time.Time
+	// If we're starting at the end of the logs, advance the start time to the most recent scan window
+	if p.config.StartPosition == End {
 		startTime = endTime.Add(-p.config.ScanFrequency)
 	}
+
+	if p.config.StartPosition == Beginning {
+		startTime = time.Unix(0, 0)
```

Comment:
Although I believe it does not really matter, setting the startTime to 0,0 sets it to `1970-01-01 00:00:00 +0000 UTC`, while the code until now was just initialising the variable in case of `beginning` which sets it to `0001-01-01 00:00:00 +0000 UTC`